spec/builtin-eval-source.test.sh

```diff
@@ -135,3 +135,23 @@ FALSE
 FALSE
 ## END
 
+#### source works for files in current directory
+echo "echo current dir" > cmd
+. cmd
+rm cmd
+## STDOUT:
+current dir
+## N-I dash stdout-json: ""
+## N-I dash status: 2
+## N-I mksh stdout-json: ""
+## N-I mksh status: 1
+
+#### source gives precendence to PATH
+mkdir -p dir
+echo "echo path" > dir/cmd
+echo "echo current dir" > cmd
+PATH="dir:$PATH"
+. cmd
+rm -r dir cmd
```

Comment:
I think `cd $TMP` is the better thing to do, but maybe I should just change the test framework to automatically do it.

Filed #396